### PR TITLE
feat: rxjava2 feature

### PR DESCRIPTION
**Pull request by sdelamo**
See https://github.com/micronaut-projects/micronaut-core/pull/5976
See https://github.com/micronaut-projects/micronaut-core/pull/5977

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java [hunk 1]
```diff
@@ -30,12 +37,37 @@
 @Singleton
 public class HttpClient implements DefaultFeature {
 
+    private final RxJavaTwoHttpClient rxJavaTwoHttpClient;
+    private final RxJavaThreeHttpClient rxJavaThreeHttpClient;
```
Comment (ilopmar) [inline starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java]: Can we rename the classes to `RxJava2HttpClient`, RxJava3HttpClient` (and also the other one `RxJava1HttpClient`

File: starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java [hunk 2]
```diff
@@ -30,12 +37,37 @@
...
+        if (featureContext.isPresent(RxJavaTwo.class)) {
+            featureContext.addFeature(rxJavaTwoHttpClient);
+        }
+        if (featureContext.isPresent(RxJavaThree.class)) {
+            featureContext.addFeature(rxJavaThreeHttpClient);
+        }
+        if (featureContext.isPresent(Reactor.class)) {
+            featureContext.addFeature(reactorHttpClient);
+        }
+    }
```
Comment (ilopmar) [inline starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java]: I think we should only allow to add one of these features. It doesn't make sense to have rxjava1, 2, 3 and reactor on the same app. Specially for RxJava.

**Review by ilopmar — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java]: It does not matter right?

File: starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java (hunk 2 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java]: But it should be part of a different PR right? If you add RxJava2 and you have other RxJava3 throw an exception.

File: starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java (hunk 1 above)
Comment (ilopmar) [reply starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java]: No, we can name the classes whatever we want but I think we should be consistent with the names. We've always named them with numbers not with letters.

File: starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java (hunk 2 above)
Comment (ilopmar) [reply starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java]: I don't see why. As part of adding a new feature we need to check the other features that are not compatible with the new one.

File: starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/other/HttpClient.java]: The class `RxJavaThree` already existed. That it is the reason I closed `RxJavaThreeHttpClient`. I can rename every class if you feel strong about this.

**Review by ilopmar — APPROVED** [review]

**Comment by jameskleeh** [comment]
I changed the PR so the features for rxjava2-http-client get added from rxjava2 instead of from http-client. This is better because the rxjava2 related features and their logic are self contained

**Pull request merged**